test/e2e/tests/update-modal/update-modal.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { version } from '../../../../package.json';
+
+describe('Update modal', function (this: Suite) {
+  it('should not be shown by default', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }) => {
+        await loginWithBalanceValidation(driver);
+        const updateModal = new UpdateModal(driver);
+        await updateModal.check_pageIsNotPresent();
+      },
+    );
+  });
+
+  it('should be shown if an update is available on an outdated version', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withAppStateController({
+            isUpdateAvailable: true,
+          })
+          .build(),
+        title: this.test?.fullTitle(),
+        manifestFlags: {
+          remoteFeatureFlags: {
+            extensionMinimumVersion: semver.inc(version, 'patch'),
+          },
+        },
+      },
+      async ({ driver }) => {
+        await loginWithBalanceValidation(driver);
+        const updateModal = new UpdateModal(driver);
+        await updateModal.check_pageIsLoaded();
+      },
+    );
+  });
+
+  it('should disappear when closed', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withAppStateController({
+            isUpdateAvailable: true,
+          })
+          .build(),
+        title: this.test?.fullTitle(),
+        manifestFlags: {
+          remoteFeatureFlags: {
+            extensionMinimumVersion: semver.inc(version, 'patch'),
+          },
+        },
+      },
+      async ({ driver }) => {
+        await loginWithBalanceValidation(driver);
+        const updateModal = new UpdateModal(driver);
+        await updateModal.check_pageIsLoaded();
+        await updateModal.close();
+        await updateModal.check_pageIsNotPresent();
+      },
+    );
+  });
+
+  it('should reload the extension when confirmed', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withAppStateController({
+            isUpdateAvailable: true,
+          })
+          .build(),
+        title: this.test?.fullTitle(),
+        disableServerMochaToBackground: true,
+        manifestFlags: {
+          remoteFeatureFlags: {
+            extensionMinimumVersion: semver.inc(version, 'patch'),
+          },
+        },
+        // we need to mock the updating page that is opened when the user confirms the update
+        testSpecificMock: async (server: Mockttp) => {
+          await server
+            .forGet('https://metamask.io/updating')
+            .thenCallback(() => ({
+              statusCode: 200,
+              body: '<title>MetaMask Updating</title>',
+            }));
+          await server
+            .forGet('https://metamask.io/favicon.ico')
+            .thenCallback(() => ({ statusCode: 200, body: '' }));
```

Comment:
Alternatively, you can add `<link rel="icon" href="data:image/png;base64,iVBORw0KGgo=">` after the `<title>` element in the mock response for `https://metamask.io/updating`.